### PR TITLE
Use Xcode 14.2 on CI

### PR DESCRIPTION
**Pull request by crazytonyli**
## Changes

What says in the title.

## Test instructions

CI checks should be enough. Nothing else I can think of.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: .xcversion [hunk 1]
```diff
@@ -1 +1 @@
-13.4.1
+14.2
```
Comment (spencertransier) [inline .xcversion:1]: Do we also increase this version each time we update the CI image in the pipelines? I'm not familiar with how we use `.xcversion`, so I wasn't sure.

**Review by spencertransier — APPROVED** [review]
Approved so you're not blocked, but I was curious about upping the `.xcversion` and how that is used in case there are implications for developers.

File: .xcversion (hunk 1 above)
Comment (crazytonyli) [reply .xcversion:1]: I wasn't sure either 🥲 . But we probably should apply the same changes in [this Woo iOS PR](https://github.com/woocommerce/woocommerce-ios/pull/8033) to this repository.

Comment (mokagio) [reply .xcversion:1]: See https://github.com/wordpress-mobile/WordPress-iOS/pull/19834

**Pull request merged**